File: fixtures/scale/kotlin/medium/src/calc/history/HistoryFormatter.kt
package calc.history

private const val DEFAULT_COUNT = 10

fun formatHistory(history: History): String =
    formatRecent(history, DEFAULT_COUNT)

fun formatRecent(history: History, count: Int): String {
    val lines = history.recent(count).map { it.summary() }
    return if (lines.isEmpty()) "(empty)" else lines.joinToString("\n")
}

File: fixtures/scale/scala/medium/src/calc/history/HistoryFormatter.scala
package calc.history

object HistoryFormatter {
  private val DefaultCount = 10

  def format(history: History): String = formatRecent(history, DefaultCount)

  def formatRecent(history: History, count: Int): String = {
    val lines = history.recent(count).map(_.summary)
    if (lines.isEmpty) "(empty)" else lines.mkString("\n")
  }
}
